`/api/tasks/TASK_ID/state` gives 500

**Describe the bug**

Just tried https://github.com/galaxyproject/gxyarchiver and got 2 problems

- No export is created (I also tried via web UI and it works) 
- Got a 500 and a long traceback in the logs:

```
uvicorn.access INFO 2024-04-02 13:22:27,620 [pN:main.2,p:4131762,tN:MainThread] 141.65.254.225:0 - "POST /api/histories/86e50fc04e8f1350/write_store HTTP/1.0" 200
uvicorn.access INFO 2024-04-02 13:22:28,344 [pN:main.2,p:4131762,tN:MainThread] 141.65.254.225:0 - "GET /history/current_history_json?since=2024-04-02T10:15:29.724974 HTTP/1.0" 200
uvicorn.access INFO 2024-04-02 13:22:28,874 [pN:main.1,p:4131758,tN:MainThread] 141.65.254.225:0 - "GET /api/tasks/8b720767-eb55-4df8-8a8e-ea20e2912b4d/state HTTP/1.0" 500
[2024-04-02 13:22:28 +0200] [4131758] [ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/anyio/streams/memory.py", line 98, in receive
    return self.receive_nowait()
           ^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/anyio/streams/memory.py", line 93, in receive_nowait
    raise WouldBlock
anyio.WouldBlock

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 78, in call_next
    message = await recv_stream.receive()
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/anyio/streams/memory.py", line 118, in receive
    raise EndOfStream
anyio.EndOfStream

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/uvicorn/protocols/http/h11_impl.py", line 428, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/fastapi/applications.py", line 284, in __call__
    await super().__call__(scope, receive, send)
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/starlette_context/middleware/raw_middleware.py", line 93, in __call__
    await self.app(scope, receive, send_wrapper)
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 108, in __call__
    response = await self.dispatch_func(request, call_next)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/lib/galaxy/webapps/galaxy/fast_app.py", line 107, in add_x_frame_options
    response = await call_next(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 84, in call_next
    raise app_exc
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 70, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
               ^^^^^^^^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/fastapi/routing.py", line 241, in app
    raw_response = await run_endpoint_function(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/fastapi/routing.py", line 169, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/starlette/concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/anyio/to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
           ^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 807, in run
    result = context.run(func, *args)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/lib/galaxy/webapps/galaxy/api/tasks.py", line 31, in state
    return self.manager.get_state(task_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/lib/galaxy/managers/tasks.py", line 76, in get_state
    return cast(TaskState, str(self._get_result(task_uuid).state))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/celery/result.py", line 503, in state
    return self._get_task_meta()['status']
           ^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/celery/result.py", line 442, in _get_task_meta
    return self._maybe_set_cache(self.backend.get_task_meta(self.id))
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs1/data/galaxy_server/galaxy-dev/.venv/lib/python3.11/site-packages/celery/backends/base.py", line 608, in get_task_meta
    meta = self._get_task_meta_for(task_id)
           ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'DisabledBackend' object has no attribute '_get_task_meta_for'
```

I have no celery setup on my Galaxy. Or better: I have not changed the default setup (I have a celery process but no idea what it is doing).

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: 23.2
Commit: a8727a32db

**To Reproduce**
Steps to reproduce the behavior:

Not sure yet. So far I just followed the 

**Expected behavior**

No error :) .. or more useful error message.

--- a/doc/source/admin/galaxy_options.rst
+++ b/doc/source/admin/galaxy_options.rst
@@ -5156,8 +5156,7 @@
     to set the `result_backend` option in the `celery_conf` option to
     a valid Celery result backend URL. By default, Galaxy uses an
     SQLite database at '<data_dir>/results.sqlite' for storing task
-    results. Please use a more robust backend (e.g. Redis) for
-    production setups. For details, see
+    results. For details, see
     https://docs.galaxyproject.org/en/master/admin/production.html#use-celery-for-asynchronous-tasks
 :Default: ``false``
 :Type: bool
@@ -5175,7 +5174,8 @@
     The `broker_url` option, if unset or null, defaults to the value
     of `amqp_internal_connection`. The `result_backend` option, if
     unset or null, defaults to an SQLite database at
-    '<data_dir>/results.sqlite' for storing task results.
+    '<data_dir>/results.sqlite' for storing task results. Please use a
+    more robust backend (e.g. Redis) for production setups.
     The galaxy.fetch_data task can be disabled by setting its route to
     "disabled": `galaxy.fetch_data: disabled`. (Other tasks cannot be
     disabled on a per-task basis at this time.)

--- a/doc/source/admin/galaxy_options.rst
+++ b/doc/source/admin/galaxy_options.rst
@@ -5154,7 +5154,9 @@
     only if you have setup a Celery worker for Galaxy and you have
     configured the `celery_conf` option below. Specifically, you need
     to set the `result_backend` option in the `celery_conf` option to
-    a valid Celery result backend URL. For details, see
+    a valid Celery result backend URL. By default a SQLite database is
+    used for storing task results, please use a more robust backend
+    for production setups like Redis. For details, see
     https://docs.galaxyproject.org/en/master/admin/production.html#use-celery-for-asynchronous-tasks
 :Default: ``false``
 :Type: bool
@@ -5177,7 +5179,7 @@
     disabled on a per-task basis at this time.)
     For details, see Celery documentation at
     https://docs.celeryq.dev/en/stable/userguide/configuration.html.
-:Default: ``{'result_backend': 'redis://127.0.0.1:6379/0', 'task_routes': {'galaxy.fetch_data': 'galaxy.external', 'galaxy.set_job_metadata': 'galaxy.external'}}``
+:Default: ``{'result_backend': 'db+sqlite:///./database/results.sqlite?isolation_level=IMMEDIATE', 'task_routes': {'galaxy.fetch_data': 'galaxy.external', 'galaxy.set_job_metadata': 'galaxy.external'}}``
 :Type: any
 
 

--- a/doc/source/admin/galaxy_options.rst
+++ b/doc/source/admin/galaxy_options.rst
@@ -5172,16 +5172,16 @@
     To refer to a task by name, use the template `galaxy.foo` where
     `foo` is the function name of the task defined in the
     galaxy.celery.tasks module.
-    The `broker_url` option, if unset, defaults to the value of
-    `amqp_internal_connection`. The `result_backend` option, if unset,
-    defaults to an SQLite database at '<data_dir>/results.sqlite' for
-    storing task results.
+    The `broker_url` option, if unset or null, defaults to the value
+    of `amqp_internal_connection`. The `result_backend` option, if
+    unset or null, defaults to an SQLite database at
+    '<data_dir>/results.sqlite' for storing task results.
     The galaxy.fetch_data task can be disabled by setting its route to
     "disabled": `galaxy.fetch_data: disabled`. (Other tasks cannot be
     disabled on a per-task basis at this time.)
     For details, see Celery documentation at
     https://docs.celeryq.dev/en/stable/userguide/configuration.html.
-:Default: ``{'task_routes': {'galaxy.fetch_data': 'galaxy.external', 'galaxy.set_job_metadata': 'galaxy.external'}}``
+:Default: ``{'broker_url': None, 'result_backend': None, 'task_routes': {'galaxy.fetch_data': 'galaxy.external', 'galaxy.set_job_metadata': 'galaxy.external'}}``
 :Type: any
 
 

--- a/doc/source/admin/galaxy_options.rst
+++ b/doc/source/admin/galaxy_options.rst
@@ -5173,14 +5173,15 @@
     `foo` is the function name of the task defined in the
     galaxy.celery.tasks module.
     The `broker_url` option, if unset, defaults to the value of
-    `amqp_internal_connection`. The `result_backend` option must be
-    set if the `enable_celery_tasks` option is set.
+    `amqp_internal_connection`. The `result_backend` option, if unset,
+    defaults to an SQLite database at '<data_dir>/results.sqlite' for
+    storing task results.
     The galaxy.fetch_data task can be disabled by setting its route to
     "disabled": `galaxy.fetch_data: disabled`. (Other tasks cannot be
     disabled on a per-task basis at this time.)
     For details, see Celery documentation at
     https://docs.celeryq.dev/en/stable/userguide/configuration.html.
-:Default: ``{'result_backend': 'db+sqlite:///./database/results.sqlite?isolation_level=IMMEDIATE', 'task_routes': {'galaxy.fetch_data': 'galaxy.external', 'galaxy.set_job_metadata': 'galaxy.external'}}``
+:Default: ``{'task_routes': {'galaxy.fetch_data': 'galaxy.external', 'galaxy.set_job_metadata': 'galaxy.external'}}``
 :Type: any
 
 

--- a/doc/source/admin/galaxy_options.rst
+++ b/doc/source/admin/galaxy_options.rst
@@ -5154,9 +5154,10 @@
     only if you have setup a Celery worker for Galaxy and you have
     configured the `celery_conf` option below. Specifically, you need
     to set the `result_backend` option in the `celery_conf` option to
-    a valid Celery result backend URL. By default a SQLite database is
-    used for storing task results, please use a more robust backend
-    for production setups like Redis. For details, see
+    a valid Celery result backend URL. By default, Galaxy uses an
+    SQLite database at '<data_dir>/results.sqlite' for storing task
+    results. Please use a more robust backend for production setups
+    like Redis. For details, see
     https://docs.galaxyproject.org/en/master/admin/production.html#use-celery-for-asynchronous-tasks
 :Default: ``false``
 :Type: bool

--- a/doc/source/admin/galaxy_options.rst
+++ b/doc/source/admin/galaxy_options.rst
@@ -5156,8 +5156,8 @@
     to set the `result_backend` option in the `celery_conf` option to
     a valid Celery result backend URL. By default, Galaxy uses an
     SQLite database at '<data_dir>/results.sqlite' for storing task
-    results. Please use a more robust backend for production setups
-    like Redis. For details, see
+    results. Please use a more robust backend (e.g. Redis) for
+    production setups. For details, see
     https://docs.galaxyproject.org/en/master/admin/production.html#use-celery-for-asynchronous-tasks
 :Default: ``false``
 :Type: bool

--- a/lib/galaxy/config/__init__.py
+++ b/lib/galaxy/config/__init__.py
@@ -1082,6 +1082,9 @@ class GalaxyAppConfiguration(BaseAppConfiguration, CommonConfigurationMixin):
             self.amqp_internal_connection = (
                 f"sqlalchemy+sqlite:///{self._in_data_dir('control.sqlite')}?isolation_level=IMMEDIATE"
             )
+
+        self._process_celery_config()
+
         self.pretty_datetime_format = expand_pretty_datetime_format(self.pretty_datetime_format)
         try:
             with open(self.user_preferences_extra_conf_path) as stream:
@@ -1202,6 +1205,16 @@ class GalaxyAppConfiguration(BaseAppConfiguration, CommonConfigurationMixin):
                 _load_theme(file_path, self.themes_by_host[host])
         else:
             _load_theme(self.themes_config_file, self.themes)
+
+    def _process_celery_config(self):
+        if self.celery_conf:
+            result_backend = self.celery_conf.get("result_backend")
+            if result_backend:
+                # If the result_backend is the default SQLite database, we need to
+                # ensure that the correct data directory is used.
+                if "results.sqlite" in result_backend:
+                    result_backend = f"db+sqlite:///{self._in_data_dir('results.sqlite')}?isolation_level=IMMEDIATE"
+                self.celery_conf["result_backend"] = result_backend
 
     def _check_database_connection_strings(self):
         """

--- a/lib/galaxy/config/__init__.py
+++ b/lib/galaxy/config/__init__.py
@@ -1207,14 +1207,10 @@ class GalaxyAppConfiguration(BaseAppConfiguration, CommonConfigurationMixin):
             _load_theme(self.themes_config_file, self.themes)
 
     def _process_celery_config(self):
-        if self.celery_conf:
-            result_backend = self.celery_conf.get("result_backend")
-            if result_backend:
-                # If the result_backend is the default SQLite database, we need to
-                # ensure that the correct data directory is used.
-                if "results.sqlite" in result_backend:
-                    result_backend = f"db+sqlite:///{self._in_data_dir('results.sqlite')}?isolation_level=IMMEDIATE"
-                self.celery_conf["result_backend"] = result_backend
+        if self.celery_conf and self.celery_conf.get("result_backend") is None:
+            # If the result_backend is not set, use a SQLite database in the data directory
+            result_backend = f"db+sqlite:///{self._in_data_dir('results.sqlite')}?isolation_level=IMMEDIATE"
+            self.celery_conf["result_backend"] = result_backend
 
     def _check_database_connection_strings(self):
         """

--- a/lib/galaxy/config/sample/galaxy.yml.sample
+++ b/lib/galaxy/config/sample/galaxy.yml.sample
@@ -2773,15 +2773,15 @@ galaxy:
   # `foo` is the function name of the task defined in the
   # galaxy.celery.tasks module.
   # The `broker_url` option, if unset, defaults to the value of
-  # `amqp_internal_connection`. The `result_backend` option must be set
-  # if the `enable_celery_tasks` option is set.
+  # `amqp_internal_connection`. The `result_backend` option, if unset,
+  # defaults to an SQLite database at '<data_dir>/results.sqlite' for
+  # storing task results.
   # The galaxy.fetch_data task can be disabled by setting its route to
   # "disabled": `galaxy.fetch_data: disabled`. (Other tasks cannot be
   # disabled on a per-task basis at this time.)
   # For details, see Celery documentation at
   # https://docs.celeryq.dev/en/stable/userguide/configuration.html.
   #celery_conf:
-  #  result_backend: db+sqlite:///./database/results.sqlite?isolation_level=IMMEDIATE
   #  task_routes:
   #    galaxy.fetch_data: galaxy.external
   #    galaxy.set_job_metadata: galaxy.external

--- a/lib/galaxy/config/sample/galaxy.yml.sample
+++ b/lib/galaxy/config/sample/galaxy.yml.sample
@@ -2763,7 +2763,6 @@ galaxy:
   # set the `result_backend` option in the `celery_conf` option to a
   # valid Celery result backend URL. By default, Galaxy uses an SQLite
   # database at '<data_dir>/results.sqlite' for storing task results.
-  # Please use a more robust backend (e.g. Redis) for production setups.
   # For details, see
   # https://docs.galaxyproject.org/en/master/admin/production.html#use-celery-for-asynchronous-tasks
   #enable_celery_tasks: false
@@ -2775,7 +2774,8 @@ galaxy:
   # The `broker_url` option, if unset or null, defaults to the value of
   # `amqp_internal_connection`. The `result_backend` option, if unset or
   # null, defaults to an SQLite database at '<data_dir>/results.sqlite'
-  # for storing task results.
+  # for storing task results. Please use a more robust backend (e.g.
+  # Redis) for production setups.
   # The galaxy.fetch_data task can be disabled by setting its route to
   # "disabled": `galaxy.fetch_data: disabled`. (Other tasks cannot be
   # disabled on a per-task basis at this time.)

--- a/lib/galaxy/config/sample/galaxy.yml.sample
+++ b/lib/galaxy/config/sample/galaxy.yml.sample
@@ -2772,16 +2772,18 @@ galaxy:
   # To refer to a task by name, use the template `galaxy.foo` where
   # `foo` is the function name of the task defined in the
   # galaxy.celery.tasks module.
-  # The `broker_url` option, if unset, defaults to the value of
-  # `amqp_internal_connection`. The `result_backend` option, if unset,
-  # defaults to an SQLite database at '<data_dir>/results.sqlite' for
-  # storing task results.
+  # The `broker_url` option, if unset or null, defaults to the value of
+  # `amqp_internal_connection`. The `result_backend` option, if unset or
+  # null, defaults to an SQLite database at '<data_dir>/results.sqlite'
+  # for storing task results.
   # The galaxy.fetch_data task can be disabled by setting its route to
   # "disabled": `galaxy.fetch_data: disabled`. (Other tasks cannot be
   # disabled on a per-task basis at this time.)
   # For details, see Celery documentation at
   # https://docs.celeryq.dev/en/stable/userguide/configuration.html.
   #celery_conf:
+  #  broker_url: null
+  #  result_backend: null
   #  task_routes:
   #    galaxy.fetch_data: galaxy.external
   #    galaxy.set_job_metadata: galaxy.external

--- a/lib/galaxy/config/sample/galaxy.yml.sample
+++ b/lib/galaxy/config/sample/galaxy.yml.sample
@@ -2763,7 +2763,7 @@ galaxy:
   # set the `result_backend` option in the `celery_conf` option to a
   # valid Celery result backend URL. By default, Galaxy uses an SQLite
   # database at '<data_dir>/results.sqlite' for storing task results.
-  # Please use a more robust backend for production setups like Redis.
+  # Please use a more robust backend (e.g. Redis) for production setups.
   # For details, see
   # https://docs.galaxyproject.org/en/master/admin/production.html#use-celery-for-asynchronous-tasks
   #enable_celery_tasks: false

--- a/lib/galaxy/config/sample/galaxy.yml.sample
+++ b/lib/galaxy/config/sample/galaxy.yml.sample
@@ -2761,7 +2761,9 @@ galaxy:
   # only if you have setup a Celery worker for Galaxy and you have
   # configured the `celery_conf` option below. Specifically, you need to
   # set the `result_backend` option in the `celery_conf` option to a
-  # valid Celery result backend URL. For details, see
+  # valid Celery result backend URL. By default a SQLite database is
+  # used for storing task results, please use a more robust backend for
+  # production setups like Redis. For details, see
   # https://docs.galaxyproject.org/en/master/admin/production.html#use-celery-for-asynchronous-tasks
   #enable_celery_tasks: false
 
@@ -2778,7 +2780,7 @@ galaxy:
   # For details, see Celery documentation at
   # https://docs.celeryq.dev/en/stable/userguide/configuration.html.
   #celery_conf:
-  #  result_backend: redis://127.0.0.1:6379/0
+  #  result_backend: db+sqlite:///./database/results.sqlite?isolation_level=IMMEDIATE
   #  task_routes:
   #    galaxy.fetch_data: galaxy.external
   #    galaxy.set_job_metadata: galaxy.external

--- a/lib/galaxy/config/sample/galaxy.yml.sample
+++ b/lib/galaxy/config/sample/galaxy.yml.sample
@@ -2761,9 +2761,10 @@ galaxy:
   # only if you have setup a Celery worker for Galaxy and you have
   # configured the `celery_conf` option below. Specifically, you need to
   # set the `result_backend` option in the `celery_conf` option to a
-  # valid Celery result backend URL. By default a SQLite database is
-  # used for storing task results, please use a more robust backend for
-  # production setups like Redis. For details, see
+  # valid Celery result backend URL. By default, Galaxy uses an SQLite
+  # database at '<data_dir>/results.sqlite' for storing task results.
+  # Please use a more robust backend for production setups like Redis.
+  # For details, see
   # https://docs.galaxyproject.org/en/master/admin/production.html#use-celery-for-asynchronous-tasks
   #enable_celery_tasks: false
 

--- a/lib/galaxy/config/schemas/config_schema.yml
+++ b/lib/galaxy/config/schemas/config_schema.yml
@@ -3772,7 +3772,6 @@ mapping:
         type: any
         required: false
         default:
-          result_backend: db+sqlite:///./database/results.sqlite?isolation_level=IMMEDIATE
           task_routes:
             'galaxy.fetch_data': 'galaxy.external'
             'galaxy.set_job_metadata': 'galaxy.external'
@@ -3783,7 +3782,8 @@ mapping:
           of the task defined in the galaxy.celery.tasks module.
 
           The `broker_url` option, if unset, defaults to the value of `amqp_internal_connection`.
-          The `result_backend` option must be set if the `enable_celery_tasks` option is set.
+          The `result_backend` option, if unset, defaults to an SQLite database at '<data_dir>/results.sqlite'
+          for storing task results.
 
           The galaxy.fetch_data task can be disabled by setting its route to "disabled": `galaxy.fetch_data: disabled`.
           (Other tasks cannot be disabled on a per-task basis at this time.)

--- a/lib/galaxy/config/schemas/config_schema.yml
+++ b/lib/galaxy/config/schemas/config_schema.yml
@@ -3772,6 +3772,8 @@ mapping:
         type: any
         required: false
         default:
+          broker_url: null
+          result_backend: null
           task_routes:
             'galaxy.fetch_data': 'galaxy.external'
             'galaxy.set_job_metadata': 'galaxy.external'
@@ -3781,8 +3783,8 @@ mapping:
           To refer to a task by name, use the template `galaxy.foo` where `foo` is the function name
           of the task defined in the galaxy.celery.tasks module.
 
-          The `broker_url` option, if unset, defaults to the value of `amqp_internal_connection`.
-          The `result_backend` option, if unset, defaults to an SQLite database at '<data_dir>/results.sqlite'
+          The `broker_url` option, if unset or null, defaults to the value of `amqp_internal_connection`.
+          The `result_backend` option, if unset or null, defaults to an SQLite database at '<data_dir>/results.sqlite'
           for storing task results.
 
           The galaxy.fetch_data task can be disabled by setting its route to "disabled": `galaxy.fetch_data: disabled`.

--- a/lib/galaxy/config/schemas/config_schema.yml
+++ b/lib/galaxy/config/schemas/config_schema.yml
@@ -3765,7 +3765,6 @@ mapping:
           configured the `celery_conf` option below. Specifically, you need to set the
           `result_backend` option in the `celery_conf` option to a valid Celery result
           backend URL. By default, Galaxy uses an SQLite database at '<data_dir>/results.sqlite' for storing task results.
-          Please use a more robust backend (e.g. Redis) for production setups.
           For details, see https://docs.galaxyproject.org/en/master/admin/production.html#use-celery-for-asynchronous-tasks
           
       celery_conf:
@@ -3785,7 +3784,7 @@ mapping:
 
           The `broker_url` option, if unset or null, defaults to the value of `amqp_internal_connection`.
           The `result_backend` option, if unset or null, defaults to an SQLite database at '<data_dir>/results.sqlite'
-          for storing task results.
+          for storing task results. Please use a more robust backend (e.g. Redis) for production setups.
 
           The galaxy.fetch_data task can be disabled by setting its route to "disabled": `galaxy.fetch_data: disabled`.
           (Other tasks cannot be disabled on a per-task basis at this time.)

--- a/lib/galaxy/config/schemas/config_schema.yml
+++ b/lib/galaxy/config/schemas/config_schema.yml
@@ -3765,7 +3765,7 @@ mapping:
           configured the `celery_conf` option below. Specifically, you need to set the
           `result_backend` option in the `celery_conf` option to a valid Celery result
           backend URL. By default, Galaxy uses an SQLite database at '<data_dir>/results.sqlite' for storing task results.
-          Please use a more robust backend for production setups like Redis.
+          Please use a more robust backend for production setups (e.g. Redis).
           For details, see https://docs.galaxyproject.org/en/master/admin/production.html#use-celery-for-asynchronous-tasks
           
       celery_conf:

--- a/lib/galaxy/config/schemas/config_schema.yml
+++ b/lib/galaxy/config/schemas/config_schema.yml
@@ -3765,7 +3765,7 @@ mapping:
           configured the `celery_conf` option below. Specifically, you need to set the
           `result_backend` option in the `celery_conf` option to a valid Celery result
           backend URL. By default, Galaxy uses an SQLite database at '<data_dir>/results.sqlite' for storing task results.
-          Please use a more robust backend for production setups (e.g. Redis).
+          Please use a more robust backend (e.g. Redis) for production setups.
           For details, see https://docs.galaxyproject.org/en/master/admin/production.html#use-celery-for-asynchronous-tasks
           
       celery_conf:

--- a/lib/galaxy/config/schemas/config_schema.yml
+++ b/lib/galaxy/config/schemas/config_schema.yml
@@ -3764,8 +3764,8 @@ mapping:
           Activate this only if you have setup a Celery worker for Galaxy and you have
           configured the `celery_conf` option below. Specifically, you need to set the
           `result_backend` option in the `celery_conf` option to a valid Celery result
-          backend URL. By default a SQLite database is used for storing task results, please
-          use a more robust backend for production setups like Redis.
+          backend URL. By default, Galaxy uses an SQLite database at '<data_dir>/results.sqlite' for storing task results.
+          Please use a more robust backend for production setups like Redis.
           For details, see https://docs.galaxyproject.org/en/master/admin/production.html#use-celery-for-asynchronous-tasks
           
       celery_conf:

--- a/lib/galaxy/config/schemas/config_schema.yml
+++ b/lib/galaxy/config/schemas/config_schema.yml
@@ -3764,14 +3764,15 @@ mapping:
           Activate this only if you have setup a Celery worker for Galaxy and you have
           configured the `celery_conf` option below. Specifically, you need to set the
           `result_backend` option in the `celery_conf` option to a valid Celery result
-          backend URL.
+          backend URL. By default a SQLite database is used for storing task results, please
+          use a more robust backend for production setups like Redis.
           For details, see https://docs.galaxyproject.org/en/master/admin/production.html#use-celery-for-asynchronous-tasks
           
       celery_conf:
         type: any
         required: false
         default:
-          result_backend: redis://127.0.0.1:6379/0
+          result_backend: db+sqlite:///./database/results.sqlite?isolation_level=IMMEDIATE
           task_routes:
             'galaxy.fetch_data': 'galaxy.external'
             'galaxy.set_job_metadata': 'galaxy.external'

